mapper/cumulocity/c8y_translator_lib/src/serializer.rs

```diff
@@ -37,21 +36,19 @@ pub enum MeasurementStreamError {
 }
 
 impl C8yJsonSerializer {
-    pub fn new(
-        default_timestamp: DateTime<FixedOffset>,
-    ) -> Result<Self, C8yJsonSerializationError> {
-        let mut serializer = C8yJsonSerializer {
-            buffer: Vec::new(),
+    pub fn new(default_timestamp: DateTime<FixedOffset>) -> Self {
+        let capa = 1024; // XXX: Choose a capacity based on expected JSON length.
```

Comment:
How to choose what capacity to choose at run time?

Reply:
Why not? This can be done in the `from_thin_edge_json_with_timestamp` using a hint size like "the c8y json output is twice bigger than the thin-edge input).

@PradeepKiruvale you can always give a try to these kind of ideas, using `cargo bench` and the `c8y_translator_lib/benches`